engines/alcachofa/menu.cpp

```diff
@@ -0,0 +1,373 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "gui/message.h"
+#include "graphics/thumbnail.h"
+
+#include "alcachofa/alcachofa.h"
+#include "alcachofa/metaengine.h"
+#include "alcachofa/menu.h"
+#include "alcachofa/player.h"
+#include "alcachofa/script.h"
+
+using namespace Common;
+using namespace Graphics;
+
+namespace Alcachofa {
+
+static void createThumbnail(ManagedSurface &surface) {
+	surface.create(kBigThumbnailWidth, kBigThumbnailHeight, PixelFormat::createFormatRGBA32());
+}
+
+static void convertToGrayscale(ManagedSurface &surface) {
+	assert(!surface.empty());
+	assert(surface.format == PixelFormat::createFormatRGBA32());
+	uint32 rgbMask = ~(uint32(0xff) << surface.format.aShift);
+
+	for (int y = 0; y < surface.h; y++) {
+		union {
+			uint32 *pixel;
+			uint8 *components;
+		};
+		pixel = (uint32 *)surface.getBasePtr(0, y);
+		for (int x = 0; x < surface.w; x++, pixel++) {
+			*pixel &= rgbMask;
+			byte gray = (components[0] + components[1] + components[2] + components[3]) / 3;
```

Comment:
```suggestion
			byte gray = (0.29 * components[1] + 0.58 * components[2] + 0.11 *components[3]) / 3;
```
And of course, do not add alpha component :D

Reply:
The original engine used the averaging, but using luminance is of course a more correct way of calculating grayscale.